lib/encoder/src/Pololu37D.cpp

```diff
@@ -5,23 +5,55 @@
 using namespace Encoder;
 
 Pololu37D::Pololu37D(const Config &config)
-    : m_QEI(config.ChannelA, config.ChannelB, config.Index, config.quadratureEncoding),
+    : m_QEI(config.ChannelA, config.ChannelB, config.Index, GPIO::QEI::Encoding::X4_ENCODING),
       m_zeroOffsetDeg(config.offsetDeg) {}
 
 bool Pololu37D::getAngleDeg(float &angle) {
   std::scoped_lock<Mutex> lock(m_mutex);
-  angle = (m_QEI.getPulses() * m_degreesPerCount) - m_zeroOffsetDeg;
+  if (read()) {
+    // reading was succesful
+    angle = m_current_angle_deg;
+  } else {
+    return false;
+  }
   return true;
 }
 
 bool Pololu37D::getAngularVelocityDegPerSec(float &speed) {
   std::scoped_lock<Mutex> lock(m_mutex);
-  speed = m_QEI.getPulseVelocity_PulsesPerSec() * m_degreesPerCount;
+  read();
+  speed = m_anglular_velocity_deg_per_sec;
   return true;
 }
 
 bool Pololu37D::reset() {
   std::scoped_lock<Mutex> lock(m_mutex);
+  m_zeroOffsetDeg                 = 0.0;
```

Comment:
its important that this reset function does the same thing for all encoders. Look at the aeat6012 code to make sure you're doing the same thing. an example of something u do differently is that u set the offset to 0 and when aeat doesnt

Reply:
not entirely sure if I can do the exact same thing. For example, I have current_pos and previous_pos which are not in the AEAT6012 class. Also, I need to run the function QEI.reset(). I have changed how what I do to the m_zeroOffsetDeg variable to be consistent with the AEAT6012 class.